lxd/instance/drivers/driver_lxc.go

```diff
@@ -6800,7 +6876,21 @@ func (d *lxc) FileSFTP() (*sftp.Client, error) {
 
 // stopForkFile attempts to send SIGINT to forkfile then waits for it to exit.
 func (d *lxc) stopForkfile() {
-	// Make sure that when the function exits, no forkfile is running by acquiring the lock (which indicates
+	// First, check if there is a SFTP process relying on this running forkfile
+	// and if so, stop it.
+	sftpMu.Lock()
+	p, found := sftpProcesses[d.forkfileRunningLockName()]
+	if found {
```

Comment:
remember we use "happy path" style, so should be `if !found{}` and handle that case here, and then further down then it can assume it was found.